internal/tokens/model.go

```diff
@@ -0,0 +1,117 @@
+package tokens
+
+import (
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/yearn/ydaemon/common/types/common"
+)
+
+// TERC20Token contains the basic information of an ERC20 token
+type TERC20Token struct {
+	Address                   ethcommon.Address   `json:"address"`
+	UnderlyingTokensAddresses []ethcommon.Address `json:"underlyingTokensAddresses,omitempty"`
+	Name                      string              `json:"name"`
+	Symbol                    string              `json:"symbol"`
+	Price                     float64             `json:"price"`
+	Decimals                  uint64              `json:"decimals"`
+	Type                      string              `json:"type"`
+}
+
+// IsVault will return true if the token is a Yearn Vault token.
+func (t *TERC20Token) IsVault() bool {
+	return t.Type == "Yearn Vault"
+}
+
+// VaultUnderlying will return the underlying token of a vault. A vault can only and
+// will only have one underlying token.
+func (t *TERC20Token) VaultUnderlying() common.Address {
+	if len(t.UnderlyingTokensAddresses) > 0 {
+		return common.FromAddress(t.UnderlyingTokensAddresses[0])
+	}
+	return common.Address{}
```

Comment:
hmm should we try to prevent having the default `common.Address{}` into the `_tokenMap` somehow?
i didn't find any loophole when this happens in the current code but just to be safe